test/e2e/test_image.py

```diff
@@ -95,3 +96,177 @@ def test_create_with_invalid_image_file(self):
+
+        assert self._get_response_image.status_code == status_code
+        assert obtained_detail == expected_detail
+
+    def post_test_images(self) -> list[dict]:
+        """
+        Posts three images. The first two have the same entity ID, and all of them have different titles and descriptions.
+
+        :return: List of dictionaries containing the expected item data returned from a get endpoint in
+                 the form of an `ImageSchema`.
+        """
+
+        entity_id_a, entity_id_b = (str(ObjectId()) for _ in range(2))
+
+        # First item
+        image_a_id = self.post_image({**IMAGE_POST_METADATA_DATA_ALL_VALUES, "entity_id": entity_id_a}, "image.jpg")
+
+        # Second item
+        image_b_id = self.post_image(
+            {
+                **IMAGE_POST_METADATA_DATA_ALL_VALUES,
+                "entity_id": entity_id_a,
+                "title": "Essay Title",
+                "description": "A research essay.",
+            },
+            "image.jpg",
+        )
+
+        # Third item
+        image_c_id = self.post_image(
+            {
+                **IMAGE_POST_METADATA_DATA_ALL_VALUES,
+                "entity_id": entity_id_b,
+                "title": "Book Title",
+                "description": "A science book.",
+            },
+            "image.jpg",
+        )
+
+        return [
+            {
+                **IMAGE_GET_DATA_ALL_VALUES,
+                "entity_id": entity_id_a,
+                "id": image_a_id,
+                "title": "Report Title",
+                "description": "A damage report.",
+            },
+            {
+                **IMAGE_GET_DATA_ALL_VALUES,
+                "entity_id": entity_id_a,
+                "id": image_b_id,
+                "title": "Essay Title",
+                "description": "A research essay.",
+            },
+            {
+                **IMAGE_GET_DATA_ALL_VALUES,
+                "entity_id": entity_id_b,
+                "id": image_c_id,
+                "title": "Book Title",
+                "description": "A science book.",
+            },
+        ]
+
+
+class TestList(ListDSL):
+    """Tests for getting a list of images."""
+
+    def test_list_with_no_filters(self):
+        """
+        Test getting a list of all images with no filters provided.
+
+        Posts 3 images and expects it to be returned.
+        """
+
+        expected_result = self.post_test_images()
+        self.get_images()
+        self.check_get_images_success(expected_result)
+
+    def test_list_with_entity_id_filter(self):
+        """
+        Test getting a list of all images with an `entity_id` filter provided.
+
+        Posts 3 images and then filter using the `entity_id`.
+        """
+
+        expected_result = self.post_test_images()[:2]
+        self.get_images(filters={"entity_id": expected_result[0]["entity_id"]})
+        self.check_get_images_success(expected_result)
```

Comment:
```suggestion
        images = self.post_test_images()
        self.get_images(filters={"entity_id": expected_result[0]["entity_id"]})
        self.check_get_images_success(images[0])
```
Just my preference, I think its slightly easier to read.

Reply:
So should `expected_result` be renamed to `images` in all of the tests in this file, or just in this one? Also would have to look more like this, since 1. expected_result is not defined and 2. the expected result is the 1st 2 images posted, not just the 1st one
```suggestion
        images = self.post_test_images()
        self.get_images(filters={"entity_id": images[0]["entity_id"]})
        self.check_get_images_success(images[:2])
```